dspace_import.py

```diff
@@ -948,20 +1041,48 @@ def import_bundles_and_bitstreams():
     import_bundle()
     import_bitstream()
 
+def write_dict_into_json(file_name, dictionary):
+    os.makedirs(MAPPING_PATH, exist_ok=True)
+    with open(MAPPING_PATH + file_name, 'w') as data:
+        data.write(json.dumps(dictionary))
+
+def get_dict_from_json(file_name):
+    return {int(key): value for key, value in read_json(file_name, MAPPING_PATH).items()}
+
+def insert_data_into_dicts():
+    global eperson_id, group_id, metadata_schema_id, metadata_field_id, community_id, collection_id, \
+        item_id, workflowitem_id, workspaceitem_id, bitstream_id, bitstreamformat_id, bundle_id
+    eperson_id = get_dict_from_json("eperson.json")
+    group_id = get_dict_from_json("group.json")
+    metadata_schema_id = get_dict_from_json("metadataschemaregistry.json")
+    metadata_field_id = get_dict_from_json("metadatafield.json")
+    community_id = get_dict_from_json("community.json")
+    collection_id = get_dict_from_json("collection.json")
+    item_id = get_dict_from_json("item.json")
+    workspaceitem_id = get_dict_from_json("workspaceitem.json")
+    workflowitem_id = get_dict_from_json("workflowitem.json")
+    bitstreamformat_id = get_dict_from_json("bitstreamformat.json")
+    bundle_id = get_dict_from_json("bundle.json")
+    bitstream_id = get_dict_from_json("bitstream.json")
 
 # call
+#insert_data_into_dicts()
 log("Data migraton started!")
+
 # at the beginning
 read_metadata()
 read_handle()
-# not depends on the ather tables
+#not depends on the ather tables
 import_handle_with_url()
 
 # you have to call together
 import_metadata()
-# import hierarchy has to call before import group
+# # import hierarchy has to call before import group
 import_hierarchy()
 import_epersons_and_groups()
 import_licenses()
 import_bundles_and_bitstreams()
+#import_user_metadata()
```

Comment:
Do we want it commented in the main?

Reply:
Okay, I found where the problem was. I created this issue before the issue regarding the import of user_metadata. The reviewer's code for the import_user_metadata method was from a branch where this method was not yet functioning. I have now deleted it from that branch. After reviewing https://github.com/dataquest-dev/dspace-python-api/pull/49, we can merge the import_user_metadata code into this branch. In that branch you can find working code for import user_metadata.